code/modules/mining/minebot.dm

```diff
@@ -292,12 +300,76 @@
 
 /obj/item/mine_bot_upgrade/health/upgrade_bot(mob/living/simple_animal/hostile/mining_drone/M, mob/user)
 	if(M.maxHealth != initial(M.maxHealth))
-		to_chat(user, "<span class='warning'>[src] already has reinforced armor!</span>")
+		to_chat(user, "<span class='warning'>The Minebot already has reinforced armor!</span>")
 		return
 	M.maxHealth += 45
 	M.updatehealth()
+	to_chat(user, "<span class='notice'>You apply the reinforced armor on the Minebot.</span>")
 	qdel(src)
 
+//Lava-Proofing  --  Fulp-Exclusive
+
+/obj/item/mine_bot_upgrade/lavaproof
+	name = "minebot lava-proofing upgrade"
+
+/obj/item/mine_bot_upgrade/lavaproof/upgrade_bot(mob/living/simple_animal/hostile/mining_drone/M, mob/user)
+	if (M.lava_proof != 0)  //I know this is a bit hacky, but I've tried to use the same technique as the other upgrades, and it somehow just didn't want to work. I've tried several methods, but this is the only one that doesn't accept the upgrade twice. Might have to do with the fact that it's a list in weather_immunities.
+		to_chat(user, "<span class='warning'>The Minebot already has lava-proof plating installed!</span>")
+		return
+	M.weather_immunities = list("lava", "ash")
+	M.lava_proof += 1  //Should be pretty straight-forward.
```

Comment:
realistically this should be a normal assignment instead of an increment. Based on your if statement above it should never leave 0 or 1 but for performance setting to 1 instead of incrementing to 1 is probably better